### PR TITLE
Uups

### PR DESCRIPTION
**Pull request by suNrisEinMyeYes**

### CODE REVIEW

File: contracts/tokens/Cruzo1155.sol [hunk 1]
```diff
@@ -28,7 +30,7 @@ contract Cruzo1155 is ERC1155URI {
      *
      */
     function total() public view onlyOwner returns (uint256) {
```
Comment (kewka) [inline contracts/tokens/Cruzo1155.sol:32]: We can remove this view, it's not used

File: contracts/tokens/Cruzo1155.sol [hunk 2]
```diff
@@ -4,13 +4,15 @@ pragma solidity ^0.8.6;
...
-    using Counters for Counters.Counter;
-    Counters.Counter private _tokenIds;
     address public marketAddress;
 
     string public name;
     string public symbol;
 
+    mapping (uint256 => address) private tokenIdToCreator;
+
+    uint256 private lastCreated;
```
Comment (kewka) [inline contracts/tokens/Cruzo1155.sol:14]: `lastCreated` is also not used

File: contracts/tokens/Cruzo1155.sol [hunk 3]
```diff
@@ -4,13 +4,15 @@ pragma solidity ^0.8.6;
...
 
 contract Cruzo1155 is ERC1155URI {
-    using Counters for Counters.Counter;
-    Counters.Counter private _tokenIds;
     address public marketAddress;
 
     string public name;
     string public symbol;
 
+    mapping (uint256 => address) private tokenIdToCreator;
```
Comment (kewka) [inline contracts/tokens/Cruzo1155.sol:12]: This is not necessary, we already have the `creators` mapping

File: contracts/tokens/Cruzo1155.sol [hunk 4]
```diff
@@ -109,7 +116,7 @@ contract Cruzo1155 is ERC1155URI {
         bytes memory _data
     ) public onlyCreator(_tokenId) returns (uint256) {
         require(
```
Comment (kewka) [inline contracts/tokens/Cruzo1155.sol:118]: This is an unreachable check, it's already checked in the `onlyCreator` modifier

File: contracts/tokens/Cruzo1155.sol [hunk 5]
```diff
@@ -126,12 +133,12 @@ contract Cruzo1155 is ERC1155URI {
     }
 
     function uri(uint256 id) public view override returns (string memory) {
-        require(id <= _tokenIds.current(), "Cruzo1155:non existent tokenId");
+        require(tokenIdToCreator[id] != address(0), "Cruzo1155:non existent tokenId");
```
Comment (kewka) [inline contracts/tokens/Cruzo1155.sol:136]:
```suggestion
        require(creators[id] != address(0), "Cruzo1155:non existent tokenId");
```

File: contracts/tokens/Cruzo1155.sol [hunk 6]
```diff
@@ -126,12 +133,12 @@ contract Cruzo1155 is ERC1155URI {
...
         return _tokenURI(id);
     }
 
     function setTokenURI(uint256 _id, string memory _uri) public {
-        require(_id <= _tokenIds.current(), "Cruzo1155:non existent tokenId");
+        require(tokenIdToCreator[_id] != address(0), "Cruzo1155:non existent tokenId");
```
Comment (kewka) [inline contracts/tokens/Cruzo1155.sol:141]:
```suggestion
        require(creators[_id] != address(0), "Cruzo1155:non existent tokenId");
```

File: contracts/tokens/Cruzo1155.sol (hunk 1 above)
Comment (suNrisEinMyeYes) [reply contracts/tokens/Cruzo1155.sol:32]: we can turn it to external view, but some tests requires this func

File: contracts/tokens/Cruzo1155.sol (hunk 2 above)
Comment (suNrisEinMyeYes) [reply contracts/tokens/Cruzo1155.sol:14]: last created is used via total, and it is reached via tests

**Pull request closed without merging**